src/mongo/client/dbclientinterface.h

```diff
@@ -1080,7 +1081,7 @@ namespace mongo {
 
            See testDbEval() in dbclient.cpp for an example of usage.
         */
-        bool eval(const std::string &dbname, const std::string &jscode, BSONObj& info, BSONElement& retValue, BSONObj *args = 0);
+        bool eval(const std::string &dbname, const std::string &jscode, bool nolock, BSONObj& info, BSONElement& retValue, BSONObj *args = 0);
```

Comment:
So, I realize we don't offer ABI compatibility, but I'd like to try if we can, and I'd even be a little sad to issue a patch release to 1.0.0 that broke callers at the source level.

We could make it a defaulted parameter which would at least retain source compatibility.

Or, we could introduce a new function.

Thoughts?